__tests__/index.test.js

```diff
@@ -340,6 +340,40 @@ describe('GET /api/articles/:article_id (comment_count)',()=>{
 
 })
 
+describe('GET /api/articles (topic query',()=>{
+    test('return a 200 code when a request is made to the api',()=>{
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+    })
+    test('return a 200 code and allarticles filtered by query when a valid request is made to the api "mitch"',()=>{
+
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(12)
+        })
+    })
+    test('return a 200 code and all articles filtered by query when a valid request is made to the api "cats"',()=>{
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(1)
+        })
+    })
+    test('return a 200 code and all articles filtered by query when a valid request is made to the api "paper"',()=>{
+        return request(app)
+        .get('/api/articles?topic=paper')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(0)
+        })
+    })
+
```

Comment:
You need to add a test for a 404 - topic not found 👀

Reply:
Hi Jodie, Can I ask you a question as I'm a little confused... on task 11, topic=paper will return no results as it is a valid topic but not associated with any articles.  topic=pens is just not in the database.  SQL will return rowCount===0 for both (I think).  So how do I differentiate between the 2?  Paul said in the lecture to use use 2 Promises in in the controller to first check that the topic exists, as well as querying the database, so if the topic check fails, the whole thing fails... which is what I was trying first... but Rich told me not to do this... so now I don't know what to think :sweat_smile: